.upstream-tests/test/support/cuda_space_selector.h

```diff
@@ -32,6 +32,20 @@
 #define SHARED
 #endif
 
+TEST_EXEC_CHECK_DISABLE
+template<class Fun>
+__host__ __device__
+void execute_on_main_thread(Fun&& fun) {
```

Comment:
Wouldn't it be better to put this into concurrent_agents.h instead?

Reply:
Its mainly used in this file, so I believe that fits well

Reply:
But it's also used elsewhere, and it feels to me that in general usage this would be used with other features in concurrent_agents.h more frequently than with the features here. It has nothing to do with space selection, and everything with concurrent agents, so having to include this file instead of concurrent_agents.h to get it feels very weird.